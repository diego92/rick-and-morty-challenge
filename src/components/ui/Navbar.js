import * as React from "react";
import AppBar from "@mui/material/AppBar";
import Box from "@mui/material/Box";
import Toolbar from "@mui/material/Toolbar";
import Typography from "@mui/material/Typography";
import Button from "@mui/material/Button";
import IconButton from "@mui/material/IconButton";
import { MenuItem } from "@mui/material";
import { Link } from "react-router-dom";

export const Navbar = () => {
  return (
    <Box sx={{ flexGrow: 1 }}>
      <AppBar position="static">
        <Toolbar>
          <Typography
            variant="h6"
            component={Link}
            to="/"
            sx={{ flexGrow: 1 }}
            style={{ textDecoration: "none", color: "white" }}
          >
            Rick And Morty
          </Typography>
          <MenuItem component={Link} to="/characters">
            <Typography textAlign="center">CHARACTERS</Typography>
          </MenuItem>
        </Toolbar>
      </AppBar>
    </Box>
  );
};
