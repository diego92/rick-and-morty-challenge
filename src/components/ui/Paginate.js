import Pagination from "@mui/material/Pagination";
import Stack from "@mui/material/Stack";
import { Grid } from "@mui/material";

export default function Paginate({ count, handleChangePage }) {
  const handleChangePaginate = (e, page) => {
    handleChangePage(page);
  };

  return (
    <Grid
      container
      direction="row"
      justifyContent="center"
      alignItems="center"
      style={{ margin: "1rem" }}
    >
      <Stack spacing={2}>
        <Pagination
          count={count}
          color="primary"
          onChange={handleChangePaginate}
        />
      </Stack>
    </Grid>
  );
}
