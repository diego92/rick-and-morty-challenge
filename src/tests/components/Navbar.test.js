import { shallow } from "enzyme";
import { Navbar } from "../../components/ui/Navbar";

describe("Pruebas en <Navbar />", () => {
  test("debe de mostrarse correctamente", () => {
    const wrapper = shallow(<Navbar />);

    expect(wrapper).toMatchSnapshot();
  });
});
