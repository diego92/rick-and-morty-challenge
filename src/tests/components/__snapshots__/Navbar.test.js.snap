// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <Navbar /> debe de mostrarse correctamente 1`] = `
<ForwardRef(Box)
  sx={
    Object {
      "flexGrow": 1,
    }
  }
>
  <ForwardRef(AppBar)
    position="static"
  >
    <ForwardRef(Toolbar)>
      <ForwardRef(Typography)
        component={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "propTypes": Object {
              "innerRef": [Function],
              "onClick": [Function],
              "replace": [Function],
              "target": [Function],
              "to": [Function],
            },
            "render": [Function],
          }
        }
        style={
          Object {
            "color": "white",
            "textDecoration": "none",
          }
        }
        sx={
          Object {
            "flexGrow": 1,
          }
        }
        to="/"
        variant="h6"
      >
        Rick And Morty
      </ForwardRef(Typography)>
      <ForwardRef(MenuItem)
        component={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "propTypes": Object {
              "innerRef": [Function],
              "onClick": [Function],
              "replace": [Function],
              "target": [Function],
              "to": [Function],
            },
            "render": [Function],
          }
        }
        to="/characters"
      >
        <ForwardRef(Typography)
          textAlign="center"
        >
          CHARACTERS
        </ForwardRef(Typography)>
      </ForwardRef(MenuItem)>
    </ForwardRef(Toolbar)>
  </ForwardRef(AppBar)>
</ForwardRef(Box)>
`;
